### ISSUE TITLE
storage: track storage space usage for dropped tables

### ISSUE THREAD

**Issue by hzxa21** [risingwavelabs/risingwave#7498]
### Is your feature request related to a problem? Please describe.

After https://github.com/risingwavelabs/risingwave/pull/6538, we have emitted metrics for total kv size of the existing tables. The metrics exclude the already dropped tables 
https://github.com/risingwavelabs/risingwave/blob/bc43426a52dc78bc56f50e4a70ccc2bca9598826/src/storage/hummock_sdk/src/table_stats.rs#L105

However, since space reclaimation is done asynchronously in the background, the dropped tables can still take up storage space and there is a lack of observability here. For example, we recently found out the the compaction trigger doesn't consider deleted/dropped data (https://github.com/risingwavelabs/risingwave/issues/6918), which can delay the space reclaimation for dropped tables siginificantly. We can easily spot this anomaly if we have the size metrics for dropped tables.

### Describe the solution you'd like

- emit the storage size metrics for dropped tables.
- delete the metrics for dropped tables only after the metric values drop to zero.
- set up an alert if the dropped table space usage remains large (future work).

**Comment by hzxa21** [comment]
> - emit the storage size metrics for dropped tables.
> - delete the metrics for dropped tables only after the metric values drop to zero.
> 

I was wrong about the current implementation. As pointed out by @zwang28, we already emit metrics for dropped tables if their data are still in LSM and not compacted.
